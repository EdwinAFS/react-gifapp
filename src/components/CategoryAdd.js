import React, { useState } from "react";
import PropTypes from "prop-types";

export const CategoryAdd = ({setCategories}) => {

	const [categoryInput, setCategoryInput] = useState('');

	const handleChangeCategory = (e) => {
		setCategoryInput(e.target.value);
	}

	const handleSubmit = ( e ) => {
		e.preventDefault();

		if( categoryInput.trim().length < 2){
			return;
		}

		setCategories( categories => [categoryInput, ...categories] );
	}

	return (
		<form onSubmit={handleSubmit}>
			<input
				type="text"
				value={categoryInput}
				onChange={handleChangeCategory}
			/>
		</form>
	);
};

CategoryAdd.propTypes = {
	setCategories: PropTypes.func.isRequired
}