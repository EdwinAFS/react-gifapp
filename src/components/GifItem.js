import React from 'react';
import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {

	return (
		<div className='card animate__animated animate__fadeInDown'>
			<div className='body'>
				<img src={url} alt={title}/>
			</div>
			<div className='footer'>
				<p>{title}</p>
			</div>
		</div>
	);
}


GifItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired
}