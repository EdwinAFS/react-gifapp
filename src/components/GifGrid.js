import React, { Fragment } from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem'; 
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

	const { data: gifs, loading } = useFetchGifs( category );

	return (
		<Fragment>
			<h3>{category}</h3>

			{ loading && <p className='animate__animated animate__lightSpeedInLeft'>Cargando...</p>}
			<div className='grid'>
				{
					gifs.map(gif => {
						return (
							<GifItem key={gif.id} {...gif} />
						);
					})
				}
			</div>

		</Fragment>
	);

}

GifGrid.protoType = {
	category: PropTypes.string.isRequired
}