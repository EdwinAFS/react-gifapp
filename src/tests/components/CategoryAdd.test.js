import { shallow } from "enzyme";
import { CategoryAdd } from "../../components/CategoryAdd";

describe('Test para el componente CategoryAdd', () => {


	const setCategories =  jest.fn();
	let wrapper;

	beforeEach(()=>{
		jest.clearAllMocks();
		wrapper = shallow( <CategoryAdd setCategories={setCategories}/> );
	});

	test('Deberia mostrar el componente CategoryAdd', ()=>{
		expect( wrapper ).toMatchSnapshot();
	});

	test('Deberia de cambiar la caja de texto', ()=>{
		const input = wrapper.find('input');
		const value = 'Hola mundo';

		input.simulate('change',{ target:{ value } });
	});
	
	test('No deberia agregar la categoria con el submit cuando la categoria es menor a 2 caracteres ', ()=>{
		
		const form = wrapper.find('form');
		form.simulate('submit',{ preventDefault: ()=>{} });

		expect( setCategories ).not.toHaveBeenCalled();

	});

	test('Deberia de llamar setCategories y limpiar la caja de texto', ()=>{

		// ingresamos la categoria
		const input = wrapper.find('input');
		const value = 'Vikings';
		input.simulate('change',{ target:{ value } });

		// realizamos el submit
		const form = wrapper.find('form');
		form.simulate('submit',{ preventDefault: ()=>{} });

		expect( setCategories ).toHaveBeenCalled();
		expect( setCategories ).toHaveBeenCalledTimes( 1 );
		expect( setCategories ).toHaveBeenCalledWith(expect.any(Function) );

		expect( input.text().trim() ).toBe('');

	});

});