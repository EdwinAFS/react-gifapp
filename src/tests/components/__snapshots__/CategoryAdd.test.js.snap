// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test para el componente CategoryAdd Deberia mostrar el componente CategoryAdd 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
